test/distributed/_composable/fsdp/test_fully_shard_autograd.py

```diff
@@ -326,5 +328,85 @@ def optim_hook(param: nn.Parameter) -> None:
                 self.assertTrue(torch.equal(ref_param, param))
 
 
+class TestFullyShardGradientScaler(FSDPTest):
```

Comment:
I think this should not go in `test_fully_shard_autograd.py`. Grad scaler is not related to autograd.

Maybe you can put it in a separate file for now? (e.g. `test_fully_shard_grad_scaler.py`?)